### PR TITLE
Astro 1439 input

### PR DESCRIPTION
**Pull request by markacianfrani**
## Brief Description

Creates a rux-input-field component

## JIRA Link

[ASTRO-1439](https://rocketcom.atlassian.net/browse/ASTRO-1439)

## General Notes

If you can find the TRON reference in this PR, I'll buy you lunch.

## Issues and Limitations

* Built in form validation currently does not work due to an oversight in how we were planning to handle the shadow dom limitation with hidden inputs. Will be investigating [stencil-forms](https://github.com/adamdbradley/stencil-forms) for some better ideas on how to handle that. 
* All regressions pass except for the textarea sections because that's not built yet

## Types of changes

-   [x] New feature

## Checklist

-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: src/stories-next/input.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        .errorText="${args.errorText}"
+        .invalid="${args.invalid}"
+        .helpText="${args.helpText}"
+        .min="${args.min}"
+        .max="${args.max}"
+        .name="${args.name}"
+        .placeholder="${args.placeholder}"
+        ?required="${args.required}"
+        ?small="${args.small}"
+        .value="${args.value}"
```
Comment (NicDominguez) [inline src/stories-next/input.stories.mdx:40]: I am still getting use to the mdx format but I believe you are missing the type and step props here as they are not working in storybook.

Type is also missing in the other stories.

File: src/components/rux-input-field/rux-input-field.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+let id = 0
+
+@Component({
+    tag: 'rux-input-field',
+    styleUrl: 'rux-input-field.scss',
+    shadow: true,
+})
+export class RuxInputField {
+    inputId = `input-${++id}`
+    inputElement?: HTMLInputElement
```
Comment (NicDominguez) [inline src/components/rux-input-field/rux-input-field.tsx:12]: Do you need the @Element decorator here?

**Review by NicDominguez — CHANGES_REQUESTED** [review]
So organized and readable. Most impressive! Just a couple of controls that are not synced up to fix.

File: src/stories-next/input.stories.mdx (hunk 1 above)
Comment (markacianfrani) [reply src/stories-next/input.stories.mdx:40]: good catch!

File: src/components/rux-input-field/rux-input-field.tsx (hunk 2 above)
Comment (markacianfrani) [reply src/components/rux-input-field/rux-input-field.tsx]: removed the ref entirely in favor of events.

**Comment by amalyah** [comment]
Note from PR review meeting from Duncan: The request for my review is not necessary here.

**Review by Cheshire89 — COMMENTED** [review]
Is there a reason your input stories differ from the format we had in astro-components ?

**Comment by markacianfrani** [comment]
> Is there a reason your input stories differ from the format we had in astro-components ?

The new MDX stories follow the convention laid out in https://rocketcom.atlassian.net/wiki/spaces/AS11/pages/1999568934/MDX+Format+Template which was largely taken from Storybook's best practices. The idea is to isolate each variant as their own separate story.

The old layout had a lot of unnecessary states like invalid number + invalid text. They're basically the same thing.

**Review by NicDominguez — APPROVED** [review]

File: src/stories/rux-form-elements.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+      <label for="input__color">Color input</label>
+      <input class="rux-input-field" type="color" id="input__color" value="#000000">
+    </li>
+    <li class="rux-form-field">
+      <label for="input__range">Range input</label>
+      <input class="rux-input-field" type="range" id="input__range" value="10">
+    </li>
+    <li class="rux-form-field">
+      <label for="input__date">Date input</label>
+      <input class="rux-input-field" type="date" id="input__date" value="1970-01-01">
```
Comment (micahjones13) [inline src/stories/rux-form-elements.stories.tsx:113]: Tron release date?

**Review by micahjones13 — APPROVED** [review]

File: src/stories/rux-form-elements.stories.tsx (hunk 3 above)
Comment (micahjones13) [reply src/stories/rux-form-elements.stories.tsx:113]: That can't be right but it's gotta have something to do with this date lol

**Pull request merged**